src/main/java/uk/gov/hmcts/reform/wataskmonitor/services/jobs/cleanup/CleanUpJobService.java

```diff
@@ -0,0 +1,193 @@
+
+import static java.util.Collections.emptyList;
+import static uk.gov.hmcts.reform.wataskmonitor.domain.taskmonitor.JobName.TASK_CLEAN_UP;
+import static uk.gov.hmcts.reform.wataskmonitor.services.ResourceEnum.ACTIVE_PROCESS_DELETE_REQUEST;
+import static uk.gov.hmcts.reform.wataskmonitor.services.ResourceEnum.CAMUNDA_CLEAN_UP_TASK_QUERY;
+import static uk.gov.hmcts.reform.wataskmonitor.services.ResourceEnum.HISTORIC_PROCESS_DELETE_REQUEST;
+
+@Component
+@Slf4j
+@SuppressWarnings("PMD.DataflowAnomalyAnalysis")
+public class CleanUpJobService {
+
+    public static final String CAMUNDA_DATE_TIME_REQUEST_PATTERN = "yyyy-MM-dd'T'HH:mm:ss.SSSZ";
+    public final DateTimeFormatter formatter = DateTimeFormatter.ofPattern(CAMUNDA_DATE_TIME_REQUEST_PATTERN);
+
+    private final CamundaClient camundaClient;
+    private final CleanUpJobConfig cleanUpJobConfig;
+
+    @Autowired
+    public CleanUpJobService(CamundaClient camundaClient,
+                             CleanUpJobConfig cleanUpJobConfig) {
+        this.camundaClient = camundaClient;
+        this.cleanUpJobConfig = cleanUpJobConfig;
+    }
+
+    public boolean isAllowedEnvironment() {
+        log.info("{} cleanUpJobConfig: {}", TASK_CLEAN_UP.name(), cleanUpJobConfig);
+
+        if (!cleanUpJobConfig.getAllowedEnvironment()
+            .contains(cleanUpJobConfig.getEnvironment().toLowerCase(Locale.ROOT))) {
+            log.info("{} is not enabled for this environment: {}",
+                TASK_CLEAN_UP.name(), cleanUpJobConfig.getEnvironment());
+            return false;
+        }
+
+        log.info("{} is enabled for this environment: {}", TASK_CLEAN_UP.name(), cleanUpJobConfig.getEnvironment());
+        return true;
+    }
+
+    public List<HistoricCamundaTask> retrieveProcesses() {
+        List<HistoricCamundaTask> historyTasks = camundaClient.getHistoryProcesses(
+            "0",
+            cleanUpJobConfig.getCamundaMaxResults(),
+            buildSearchQuery()
+        );
+        log.info("{} task(s) retrieved successfully from history", historyTasks.size());
+        return historyTasks;
+    }
+
+    public GenericJobReport deleteHistoricProcesses(List<HistoricCamundaTask> historicCamundaTasks,
+                                                    String serviceToken) {
+        if (historicCamundaTasks == null || historicCamundaTasks.isEmpty()) {
+            log.info("There was no task(s) to delete.");
+            return new GenericJobReport(0, emptyList());
+        } else {
+            List<GenericJobOutcome> outcomesList;
+
+            outcomesList = deleteHistoricProcessesFromCamunda(historicCamundaTasks, serviceToken);
+
+            return new GenericJobReport(historicCamundaTasks.size(), outcomesList);
+        }
+    }
+
+    public GenericJobReport deleteActiveProcesses(List<HistoricCamundaTask> historicCamundaTasks, String serviceToken) {
+
+        if (!isCleanActiveTaskAllowed()) {
+            return new GenericJobReport(0, emptyList());
```

Comment:
maybe add a log here that we didn't actually run the method since it is configured off?